aspnetcore/blazor/components/integration.md

```diff
@@ -186,9 +208,162 @@ Add a `Counter` component to the app with the interactive server render mode.
+
+Delete the donor Blazor Web App, as it has no further use in this process.
+
+Add the `.Client` project to the solution:
+
+* Visual Studio: Right-click the solution in **Solution Explorer** and select **Add** > **Existing Project**. Navigate to the `.Client` folder and select the project file (`.csproj`).
+
+* .NET CLI: Use the [`dotnet sln add` command](/dotnet/core/tools/dotnet-sln#add) to add the `.Client` project to the solution.
+
+Add a project reference to the ASP.NET Core project for the client project:
+
+* Visual Studio: Right-click the ASP.NET Core project and select **Add** > **Project Reference**. Select the `.Client` project and select **OK**.
+
+* .NET CLI: From the ASP.NET Core project's folder, use the following command:
+
+  ```dotnetcli
+  dotnet add reference ../AspNetCoreApp.Client/AspNetCoreApp.Client.csproj
+  ```
+
+  The preceding command assumes the following:
+  
+  * The project file name is `AspNetCoreApp.Client.csproj`.
+  * The `.Client` project is in a `AspNetCoreApp.Client` folder inside the solution folder. The `.Client` folder is side-by-side with the ASP.NET Core project's folder.
+  
+  For more information on the `dotnet add reference` command, see [`dotnet add reference` (.NET documentation)](/dotnet/core/tools/dotnet-add-reference).
+
+Make the following changes to the ASP.NET Core app's `Program` file:
+
+* Add interactive WebAssembly component services with `AddInteractiveWebAssemblyComponents` where Razor component services are added with `AddRazorComponents`.
+
+  For interactive Auto rendering:
+
+  ```csharp
+  builder.Services.AddRazorComponents()
+      .AddInteractiveServerComponents()
+      .AddInteractiveWebAssemblyComponents();
+  ```
+
+  For only interactive WebAssembly rendering:
+
+  ```csharp
+  builder.Services.AddRazorComponents()
+      .AddInteractiveWebAssemblyComponents();
+  ```
+
+* Add the interactive WebAssembly render mode (`AddInteractiveWebAssemblyRenderMode`) and additional assemblies for the `.Client` project where Razor components are mapped with `MapRazorComponents`.
+
+  For interactive Auto rendering:
+
+  ```csharp
+  app.MapRazorComponents<App>()
+      .AddInteractiveServerRenderMode()
+      .AddInteractiveWebAssemblyRenderMode()
+      .AddAdditionalAssemblies(typeof(AspNetCoreApp.Client._Imports).Assembly);
+  ```
+
+  For only interactive WebAssembly rendering:
+
+  ```csharp
+  app.MapRazorComponents<App>()
+      .AddInteractiveWebAssemblyRenderMode()
+      .AddAdditionalAssemblies(typeof(AspNetCoreApp.Client._Imports).Assembly);
+  ```
+
+  In the preceding examples, change `AspNetCoreApp.Client` to match the `.Client` project's namespace.
+
+Add a `Components` folder and a `Components/Pages` folder to the `.Client` project.
+
+Add the following `Counter` component (`Components/Pages/Counter.razor`) to the `.Client` project:
```

Comment:
If the user already added a Counter component to the server project, then they may end up with two Counter components that have a routing conflict. If they already have a Counter can we tell them to just move it to the client, otherwise add one?

Reply:
It instructs them not to add the sample code in the earlier step.

Reply:
Lines 238 and 244.

Reply:
Do you still want me to add a line with that sentiment :point_up:? ... Move it if you have it.

Reply:
***Nevermind*** ... I forgot what we were doing with all of the churn. I'll add a remark on it :point_up:.